test/fail_compilation/fail15702.d

```diff
@@ -0,0 +1,43 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/fail15702.d(27): Error: cannot implicitly convert expression (ptrs) of type int*[] with indirections to void[] in @safe code
+fail_compilation/fail15702.d(28): Error: cannot implicitly convert expression (objs) of type Object[] with indirections to void[] in @safe code
+fail_compilation/fail15702.d(29): Error: cannot implicitly convert expression ([new Object]) of type Object[] with indirections to void[] in @safe code
+---
+*/
+void notTrustworthy(void[] buf) @trusted
+{
+    auto bytes = cast(ubyte[]) buf;
+    bytes[0] = 123;
+}
+
+int[] ints;
+int*[] ptrs;
+Object[] objs;
+
+void safeFun() @safe
+{
+    // should be allowed:
+    notTrustworthy(ints);
+    notTrustworthy(cast(void[]) ptrs);
+    notTrustworthy(cast(void[]) objs);
+
+    // should be prohibited:
+    notTrustworthy(ptrs);
+    notTrustworthy(objs);
+    void[] arr = [ new Object() ];
+}
+
+void unsafeFun() @system
```

Comment:
I would split the acceptance test in compilable.

Also, could you provide additional tests for an array of:
- struct with a static array (should be accepted);
- struct with a pointer member (rejected);
- struct with a dynamic array member (rejected);